Density/app/src/main/java/org/cornelldti/density/density/facilitydetail/FacilityInfoPage.kt

```diff
@@ -493,6 +491,16 @@ class FacilityInfoPage : BaseActivity() {
                 adapter = menuItemListViewAdapter
             }
 
+            // This is where the operating hours for the selected meal of day is set!
+            if(availableMenus.isNotEmpty() && opHoursStrings.isNotEmpty()) {
+                menuHours.text = opHoursStrings[availableMenus.indexOf(mealOfDay)]
+                clock_image.visibility = View.VISIBLE
```

Comment:
This doesn't work because we don't fetch operating hours more than once (which might be wrong to begin with, now that I think about it).

Reply:
This currently causes an out of bounds error.

Reply:
wait what about it does? i never got an out of bounds error

Reply:
yup, try something that doesn't have the same meals everyday like Becker; also, you can tell based on the code because the hours don't refresh

Reply:
> This doesn't work because we don't fetch operating hours more than once (which might be wrong to begin with, now that I think about it).

The way I've implemented it now is that it will fetch operating hours once using the operating hours endpoint. This fetch gets the current day's hours and the first time slot of tomorrow so that it can accurately display the open/closed and when its open until or next opens on the toolbar. Then there is also a fetch for operating hours everytime the menu api endpoint is hit. The operating hours are a part of the menu object now, so in the showMenu() function, it now just displays the corresponding open hours for a specific meal of day of the currently selected menu.